src/launch/java/baritone/launch/mixins/MixinBitArray.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.launch.mixins;
+
+import baritone.utils.accessor.IBitArray;
+import net.minecraft.util.BitArray;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+
+@Mixin(BitArray.class)
+public abstract class MixinBitArray implements IBitArray {
```

Comment:
how much faster is this really. maybe if this effects performance for things other then world scanner then maybe it would be worth re-licensing under MIT so it can be pull requested into paper or spigot and forge 🤔

Reply:
that is reasonable, logan's decision tho

Reply:
@0-x-2-2 it is quite a bit faster, it slowed up world scanning as a lot, the problem is that is actually breaks worldloading on forge when I overwrote the method so I had to make it an accessor instead

Reply:
> breaks worldloading on forge

ok this is epic

are there any changes you've made that could affect minecraft internal behavior? the iblockstate hashcode comes to mind -- i know that they are intended to be immutable, but couldn't some annoying forge mod developer subclass blockstatecontainer and make it mutable or something?

Reply:
> couldn't some annoying forge mod developer subclass blockstatecontainer and make it mutable or something

then the forge developer is a fucking idiot and broke minecraft and that's not my problem

Reply:
> @0-x-2-2 it is quite a bit faster, it sped up world scanning as a lot, the problem is that is actually breaks worldloading on forge when I overwrote the method so I had to make it an accessor instead

Did it speed up world loading on vanilla tho 🤔

Reply:
> Did it speed up world loading on vanilla tho 🤔

This isn't designed for world loading, it's made for the world scanner.